bridge/jira/export.go

```diff
@@ -0,0 +1,455 @@
+	}
+
+	// cache operation jira id
+	self.cachedOperationIDs[createOp.Id()] = bugJiraID
+
+	// lookup the mapping from git-bug "status" to JIRA "status" id
+	statusMap, err := getStatusMap(self.conf)
+	if err != nil {
+		return err
+	}
+
+	for _, op := range snapshot.Operations[1:] {
+		// ignore SetMetadata operations
+		if _, ok := op.(*bug.SetMetadataOperation); ok {
+			continue
+		}
+
+		// ignore operations already existing in jira (due to import or export)
+		// cache the ID of already exported or imported issues and events from
+		// Jira
+		if id, ok := op.GetMetadata(keyJiraID); ok {
+			self.cachedOperationIDs[op.Id()] = id
+			continue
+		}
+
+		opAuthor := op.GetAuthor()
+		client, err := self.getIdentityClient(ctx, opAuthor.Id())
+		if err != nil {
+			out <- core.NewExportError(
+				fmt.Errorf("missing operation author credentials for user %.8s",
+					author.Id().String()), op.Id())
+			continue
+		}
+
+		var id string
+		var exportTime time.Time
+		switch opr := op.(type) {
+		case *bug.AddCommentOperation:
+			comment, err := client.AddComment(bugJiraID, opr.Message)
+			if err != nil {
+				err := errors.Wrap(err, "adding comment")
+				out <- core.NewExportError(err, b.Id())
+				return err
+			}
+			id = comment.ID
+			out <- core.NewExportComment(op.Id())
+
+			// cache comment id
+			self.cachedOperationIDs[op.Id()] = id
+
+		case *bug.EditCommentOperation:
+			if opr.Target == createOp.Id() {
+				// An EditCommentOpreation with the Target set to the create operation
+				// encodes a modification to the long-description/summary.
+				exportTime, err = client.UpdateIssueBody(bugJiraID, opr.Message)
+				if err != nil {
+					err := errors.Wrap(err, "editing issue")
+					out <- core.NewExportError(err, b.Id())
+					return err
+				}
+				out <- core.NewExportCommentEdition(op.Id())
+				id = bugJiraID
+			} else {
+				// Otherwise it's an edit to an actual comment. A comment cannot be
+				// edited before it was created, so it must be the case that we have
+				// already observed and cached the AddCommentOperation.
+				commentID, ok := self.cachedOperationIDs[opr.Target]
+				if !ok {
+					// Since an edit has to come after the creation, we expect we would
+					// have cached the creation id.
+					panic("unexpected error: comment id not found")
+				}
+				comment, err := client.UpdateComment(bugJiraID, commentID, opr.Message)
+				if err != nil {
+					err := errors.Wrap(err, "editing comment")
+					out <- core.NewExportError(err, b.Id())
+					return err
+				}
+				out <- core.NewExportCommentEdition(op.Id())
+				// JIRA doesn't track all comment edits, they will only tell us about
+				// the most recent one. We must invent a consistent id for the operation
+				// so we use the comment ID plus the timestamp of the update, as
+				// reported by JIRA. Note that this must be consistent with the importer
+				// during ensureComment()
+				id = fmt.Sprintf("%s-%d", comment.ID, comment.Updated.Unix())
```

Comment:
I think you can use `getTimeDerivedID` here

Reply:
Done